plugin/builtin/buildbaron/build_baron.go

```diff
@@ -130,6 +152,45 @@ type searchReturnInfo struct {
 	Search string                  `json:"search"`
 }
 
+// raceSuggesters returns the JIRA ticket results from the altEndpoint suggester if it returns
+// within its configured interval, and returns the JIRA ticket results from the fallback suggester
+// otherwise.
+func raceSuggesters(fallback, altEndpoint suggester, t *task.Task) ([]thirdparty.JiraTicket, error) {
+	type result struct {
+		Tickets []thirdparty.JiraTicket
+		Error   error
+	}
+
+	// thirdparty/jira.go and thirdparty/http.go do not expose an API that accepts a context.Context.
+	fallbackCtx := context.TODO()
+	fallbackChan := make(chan result, 1)
+	go func() {
+		suggestions, err := fallback.Suggest(fallbackCtx, t)
+		fallbackChan <- result{suggestions, err}
+		close(fallbackChan)
+	}()
+
+	altEndpointTimeout := altEndpoint.GetTimeout()
+	altEndpointCtx, altEndpointCancel := context.WithTimeout(context.Background(), altEndpointTimeout)
+	defer altEndpointCancel()
+	suggestions, err := altEndpoint.Suggest(altEndpointCtx, t)
+
+	// If the alternative endpoint didn't respond quickly enough or didn't have results available,
+	// then we wait for the fallback results. Ideally we'd otherwise be able to cancel the request
+	// for fetching the fallback results, but we instead just return back to the caller without
+	// waiting for the associated goroutine to complete.
+	if err != nil {
+		grip.Warningf(
```

Comment:
We prefer structured logging since it's easier to use with splunk. Can you do change this to the following:
```
grip.Warning(message.WrapError(err, message.Fields{
"message" : "failed to get results from alternative endpoint",
"task_id": t.Id,
"execution": t.Execution,
}))
```